pkg/operator/reports.go

```diff
@@ -91,8 +91,26 @@ func (op *Reporting) syncReport(logger log.FieldLogger, key string) error {
 		}
 		return err
 	}
+
 	sr := report.DeepCopy()
 
+	// check if the report is expired
+	if report.DeletionTimestamp == nil {
+		if reportExpired := isReportExpired(logger, report, time.Now()); reportExpired {
+			if reportIsNotInput := isReportNotUsedAsInput(report, op); reportIsNotInput {
+				op.deleteReport(sr)
+				newDeletionTimestamp := metav1.Now()
+				report.SetDeletionTimestamp(&newDeletionTimestamp)
+				op.meteringClient.MeteringV1().Reports(report.Namespace).
+					Delete(context.TODO(), report.Name, metav1.DeleteOptions{})
```

Comment:
Hmm, this looks like it's deleting the actual object we grabbed from the cache instead of the deep copy that was initialized on line 95.

Reply:
`Reports().Delete` just takes the Name, so I don't think it matters?